src/silly_sprites/event_system.hpp

```diff
@@ -0,0 +1,67 @@
+#include "events.hpp"
+#include "types.hpp"
+#include <vector>
+
+namespace sly::event {
+
+    struct EventID final {
```

Comment:
I don't know what you think about this, but I would spell it `EventId` instead of `EventID` (just like `GlfwContext` instead of `GLFWContext`).